src/main/kotlin/io/github/inductiveautomation/kindling/idb/tagconfig/model/ProviderStatistics.kt

```diff
@@ -0,0 +1,135 @@
+        QuantitativeStatistic("totalUdtInstances"),
+        QuantitativeStatistic("totalUdtDefinitions"),
+        QuantitativeStatistic("totalTagsWithAlarms"),
+        QuantitativeStatistic("totalAlarms"),
+        QuantitativeStatistic("totalTagsWithHistory"),
+        QuantitativeStatistic("totalTagsWithEnabledScripts"),
+        QuantitativeStatistic("totalEnabledScripts"),
+        DependentStatistic("totalOrphanedTags", orphanedTags) { it.size },
+        QuantitativeStatistic("totalReadOnlyTags")
+    )
+
+    private val statsMap = all.associateBy { it.name }
+
+    val totalAtomicTags: QuantitativeStatistic by statsMap
+    val totalFolders: QuantitativeStatistic by statsMap
+    val totalUdtInstances: QuantitativeStatistic by statsMap
+    val totalTagsWithAlarms: QuantitativeStatistic by statsMap
+    val totalUdtDefinitions: QuantitativeStatistic by statsMap
+    val totalAlarms: QuantitativeStatistic by statsMap
+    val totalTagsWithHistory: QuantitativeStatistic by statsMap
+    val totalTagsWithEnabledScripts: QuantitativeStatistic by statsMap
+    val totalEnabledScripts: QuantitativeStatistic by statsMap
+    val dataTypes: MappedStatistic by statsMap
+    val valueSources: MappedStatistic by statsMap
+    val totalOrphanedTags: DependentStatistic<Int, MutableList<Node>> by statsMap
+    val totalReadOnlyTags: QuantitativeStatistic by statsMap
+
+    fun processNodeForStatistics(node: Node) {
+        if (node.statistics.isUdtDefinition) {
+            totalUdtDefinitions.value++
+            return
+        }
+
+        when {
+            node.statistics.isAtomicTag -> {
+                totalAtomicTags.value++
+
+                dataTypes.value.compute(node.statistics.dataType ?: DEFAULT_DATA_TYPE) { _, value ->
+                    if (value == null) 1 else value + 1
+                }
+
+                valueSources.value.compute(node.statistics.dataSource ?: DEFAULT_VALUE_SOURCE) { _, value ->
+                    if (value == null) 1 else value + 1
+                }
+            }
+
+            node.statistics.isFolder -> totalFolders.value++
+            node.statistics.isUdtInstance -> totalUdtInstances.value++
+            node.statistics.isReadOnly == true -> totalReadOnlyTags.value++
+        }
+
+        if (node.statistics.historyEnabled == true) totalTagsWithHistory.value++
+
+        if (node.statistics.hasAlarms) {
+            totalTagsWithAlarms.value++
+            totalAlarms.value += node.statistics.numAlarms
+        }
+
+        if (node.statistics.hasScripts) {
+            totalTagsWithEnabledScripts.value++
+            totalEnabledScripts.value += node.statistics.numScripts
+        }
+    }
+
+    override fun toString(): String = buildString {
+        appendLine("Atomic Tags: $totalAtomicTags")
+        appendLine("- By Data Source")
+        appendLine(valueSources.value.entries.joinToString("\n|- ", prefix = "|- ") { (key, value) -> "$key: $value" })
```

Comment:
Nit: For performance/efficiency, if you're already in a StringBuilder, prefer:
`valueSources.value.entries.joinTo(this, ...)` - so you don't allocate excess internal strings, you just keep appending to the already existing buffer